dev/AnimatedIcon/AnimatedIcon.cpp

```diff
@@ -59,17 +59,41 @@ void AnimatedIcon::OnApplyTemplate()
 
 void AnimatedIcon::OnLoaded(winrt::IInspectable const&, winrt::RoutedEventArgs const&)
 {
-    // AnimatedIcon might get added to a UI which has already set the State property on the parent.
+    // AnimatedIcon might get added to a UI which has already set the State property on the parent or grand parent.
     // If this is the case and the animated icon being added doesn't have its own state property
-    // We copy the parent value when we load.
+    // We copy the ancestor value when we load. Additionally we attach to our parent and grand parent's property
```

Comment:
Just walking one more level seems a bit fragile. If there are more element in between the icon and the presenter, then this will break. Also if the value gets set after this walk, then that will not work.